packages/klighd-core/src/proxy-view/proxy-view-actions.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * KIELER - Kiel Integrated Environment for Layout Eclipse RichClient
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ *
+ * Copyright 2022 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+import { inject, injectable } from "inversify";
+import { ActionHandlerRegistry, IActionHandler, IActionHandlerInitializer, ICommand, SetUIExtensionVisibilityAction } from "sprotty";
+import { Action, Bounds, SetModelAction, UpdateModelAction } from "sprotty-protocol";
+import { SendModelContextAction } from "../actions/actions";
+import { DISymbol } from "../di.symbols";
+import { OptionsRegistry } from "../options/options-registry";
+import { RenderOptionsRegistry } from "../options/render-options-registry";
+import { SynthesesRegistry } from "../syntheses/syntheses-registry";
+import { ProxyView } from "./proxy-view";
+
+//////// Actions ////////
+
+/** Wrapper action around {@link SetUIExtensionVisibilityAction} which shows the proxy.
+  * Otherwise the proxy-view would be invisible. */
+export type ShowProxyViewAction = SetUIExtensionVisibilityAction
+
+export namespace ShowProxyViewAction {
+    export function create(): ShowProxyViewAction {
+        return SetUIExtensionVisibilityAction.create({
+            extensionId: ProxyView.ID,
+            visible: true,
+        })
+    }
+}
+
+/** Sent from the {@link ProxyView} to the action handler to avoid Stackoverflows. */
+export interface SendProxyViewAction extends Action {
+    kind: typeof SendProxyViewAction.KIND
+    proxyView: ProxyView
+}
+
+export namespace SendProxyViewAction {
+    export const KIND = 'sendProxyViewAction'
+
+    export function create(proxyView: ProxyView): SendProxyViewAction {
+        return {
+            kind: KIND,
+            proxyView
+        }
+    }
+}
+
+/**
+ * Handles {@link SendProxyViewAction}s to get the {@link ProxyView} instance
+ * aswell as {@link SendModelContextAction}s to redirect the content to the {@link ProxyView}.
+ */
+@injectable()
+export class ProxyViewActionHandler implements IActionHandler, IActionHandlerInitializer {
+    private proxyView: ProxyView;
+    @inject(DISymbol.SynthesesRegistry) private synthesesRegistry: SynthesesRegistry;
+    @inject(DISymbol.RenderOptionsRegistry) private renderOptionsRegistry: RenderOptionsRegistry;
+    @inject(DISymbol.OptionsRegistry) private optionsRegistry: OptionsRegistry;
+    /** Whether the proxy-view was registered in the registries' onchange() method. Prevents registering multiple times. */
```

Comment:
Why can't it be registered once during initialization to avoid this altogether?

Reply:
Since there are multiple registries plural should be correct.
The registries cannot be injected in the ProxyView since it results in a circular dependency - therefore it can't be done in the ProxyView's init() method. Regarding an init() method in the ProxyViewActionHandler, the proxyView field is most likely undefined when init() is called. Therefore doing it this way is the 'safe' way, since you only register once the ProxyView is sure to be instantiated.
Although the SendProxyViewAction should only be sent once anyway, but well, better to be safe than sorry